### PR TITLE
wasm: retry if file exists

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

fixes: https://github.com/istio/istio/issues/42502

cc @ingwonsong

### CODE REVIEW

File: pkg/wasm/convert.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 	"sync"
 	"time"
 
+	"github.com/cenkalti/backoff/v4"
```
Comment (hzxuzhonghu) [inline pkg/wasm/convert.go:22]: use istio wrapped backoff

File: pkg/wasm/convert.go [hunk 2]
```diff
@@ -215,16 +217,34 @@ func convertWasmConfigFromRemoteToLocal(ec *core.TypedExtensionConfig, wasmHTTPF
...
-	})
+
+	var (
+		f        string
+		err      error
+		attempts = 0
+	)
+
+	b := backoff.NewExponentialBackOff(backoff.DefaultOption())
+	for attempts < cacheMaxTry {
```
Comment (hzxuzhonghu) [inline pkg/wasm/convert.go:228]: Suggest using RetryWithContext(), and may add MaxRetries in that lib

Comment (zirain) [reply pkg/wasm/convert.go:228]: In this case, only retry if file exists

File: pkg/wasm/convert.go [hunk 3]
```diff
@@ -215,16 +217,34 @@ func convertWasmConfigFromRemoteToLocal(ec *core.TypedExtensionConfig, wasmHTTPF
...
+		// https://github.com/istio/istio/blob/9ea7ad532a9cc58a3564143d41ac89a61aaa8058/pilot/pkg/networking/core/v1alpha3/extension/wasmplugin.go#L103
+		f, err = cache.Get(httpURI.GetUri(), GetOptions{
+			Checksum:        remote.Sha256,
+			ResourceName:    ec.Name,
+			ResourceVersion: resourceVersion,
+			RequestTimeout:  timeout,
+			PullSecret:      pullSecret,
+			PullPolicy:      pullPolicy,
+		})
+		if err != nil && strings.Contains(err.Error(), "file exists") {
```
Comment (ingwonsong) [inline pkg/wasm/convert.go:240]: String comparison for the error message seems to be not a good choice.

**Comment by zirain** [comment]
replaced by https://github.com/istio/istio/pull/43107

**Pull request closed without merging**